apps/sms/views/inbox/js/inbox.js

```diff
@@ -697,77 +704,96 @@ var InboxView = {
     });
   },
 
-  createThread: function inbox_createThread(record) {
-    // Create DOM element
-    var li = document.createElement('li');
-    var timestamp = +record.timestamp;
-    var type = record.lastMessageType;
-    var participants = record.participants;
-    var number = participants[0];
+  /**
+   * This method is responsible for adding or updating the data that could
+   * change in a node that's not moving in the same time.
+   *
+   * @param {Node} node The Node to update.
+   * @param {Object} record An object that holds the conversation informations.
+   */
+  updateConversationContent(node, record) {
     var id = record.id;
-    var bodyHTML = record.body;
-    var thread = Threads.get(id);
-    var iconLabel = '';
 
-    // A new conversation "is" a draft
+    var thread = Threads.get(id);
     var isDraft = thread.isDraft;
-
-    // A an existing conversation has draft.
     var draft = thread.getDraft();
 
-    if (!isDraft && draft) {
-      // If the draft is newer than the message, update
-      // the body with the draft content's first string.
-      if (draft.timestamp >= record.timestamp) {
-        bodyHTML = draft.content.find(function(content) {
-          if (typeof content === 'string') {
-            return true;
-          }
-        });
-        type = draft.type;
-      }
-    }
+    var body = record.body;
+    var type = record.lastMessageType;
 
-    bodyHTML = Template.escape(bodyHTML || '');
+    var iconLabel = '';
 
-    li.id = 'thread-' + id;
-    li.dataset.threadId = id;
-    li.dataset.time = timestamp;
-    li.dataset.lastMessageType = type;
-    li.classList.add('threadlist-item');
+    // If the draft is newer than the message, update
+    // the body with the draft content's first string.
+    if (draft && draft.timestamp >= record.timestamp) {
+      body = draft.content.find((content) => typeof content === 'string');
+      type = draft.type;
+    }
+
+    node.dataset.lastMessageType = type;
 
     if (draft) {
       // Set the "draft" visual indication
-      li.classList.add('draft');
+      node.classList.add('draft');
 
       if (isDraft) {
-        li.dataset.draftId = draft.id;
-        li.classList.add('is-draft');
+        node.dataset.draftId = draft.id;
+        node.classList.add('is-draft');
         iconLabel = 'is-draft';
       } else {
-        li.classList.add('has-draft');
+        node.classList.add('has-draft');
```

Comment:
__issue:__ our integration test caught that we don't remove "has-draft" css class when draft is discarded :)